src/routes/middlewares/uploadProductImages.js

```diff
@@ -0,0 +1,58 @@
+import crypto from 'crypto';
+import * as dotenv from 'dotenv';
+import Grid from 'gridfs-stream';
+import mongoose from 'mongoose';
+import multer from 'multer';
+import { GridFsStorage } from 'multer-gridfs-storage';
+import path from 'path';
+
+dotenv.config();
+
+const conn = mongoose.createConnection(process.env.DB_URI, {
+	useNewUrlParser: true,
+	useUnifiedTopology: true,
+});
+
+// Init gfs
+let gfs;
+
+conn.once('open', () => {
+	// Init stream
+	gfs = Grid(conn.db, mongoose.mongo);
+	gfs.collection('product_images'); //collection name
+});
+
+// Create storage engine
+const storage = new GridFsStorage({
+	url: process.env.DB_URI,
+	file: (req, file) => {
+		return new Promise((resolve, reject) => {
+			try {
+				crypto.randomBytes(16, (err, buf) => {
+					if (err) {
+						return reject(err);
+					}
+					const filename =
+						buf.toString('hex') + path.extname(file.originalname);
+					const fileInfo = {
+						filename: filename,
+						bucketName: 'product_images', //collection name
+					};
+
+					req.body.images = !req.body.images ? [] : req.body.images;
+					req.body.images.push({
+						altName: file.originalname,
+						name: filename,
+						src: `${process.env.SERVICE_PROXY}/products/images/${filename}`,
```

Comment:
I'm not sure what this environment variable is, and it's also not mentioned in the documentation along with the `JWT_SECRET`.

Reply:
This should be removed now. Its code from other functionality. And you don't need to worry about this anymore.